tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -515,51 +561,67 @@ final class ScalaJSClassEmitter(semantics: Semantics, outputMode: OutputMode,
     val ancestorsRecord = js.ObjectConstr(
         tree.ancestors.map(ancestor => (js.Ident(ancestor), js.IntLiteral(1))))
 
-    val typeData = js.Apply(js.New(envField("TypeData"), Nil) DOT "initClass", (List(
+    val (isInstanceFun, isArrayOfFun) = {
+      if (isObjectClass) {
+        /* Object has special ScalaJS.is.O *and* ScalaJS.isArrayOf.O. */
+        (envField("is", className), envField("isArrayOf", className))
+      } else if (isHijackedBoxedClass) {
+        /* Hijacked boxed classes have a special isInstanceOf test. */
+        val xParam = js.ParamDef(Ident("x"), rest = false)
+        (js.Function(List(xParam), js.Return {
+          genIsInstanceOf(xParam.ref, ClassType(className))
+        }), js.Undefined())
+      } else if (isAncestorOfHijackedClass || className == StringClass) {
+        /* java.lang.String and ancestors of hijacked classes have a normal
+         * ScalaJS.is.pack_Class test but with a non-standard behavior. */
+        (envField("is", className), js.Undefined())
+      } else if (tree.kind == ClassKind.RawJSType) {
+        /* Raw JS types have an instanceof operator-based isInstanceOf test
+         * dictated by their jsName. If there is no jsName, the test cannot
+         * be performed and must throw.
+         */
+        tree.jsName.fold[(js.Tree, js.Tree)] {
+          (envField("noIsInstance"), js.Undefined())
```

Comment:
It is so that `classOf[SomeRawJSTrait].isInstance(obj)` throws, instead of returning `true` or `false`. It works for a raw JS class, or anything else.